src/Application/Responses/JsonResponse.php

```diff
@@ -26,18 +26,23 @@ class JsonResponse extends Nette\Object implements Nette\Application\IResponse
 	/** @var string */
 	private $contentType;
 
+	/** @var string */
+	private $charset;
+
 
 	/**
 	 * @param  array|\stdClass  payload
-	 * @param  string    MIME content type
+	 * @param  string           MIME content type
+	 * @param  string           charset
 	 */
-	public function __construct($payload, $contentType = NULL)
+	public function __construct($payload, $contentType = NULL, $charset = NULL)
```

Comment:
Oh, I did not realize the due to using `json_encode` the encoding will always be UTF-8.